docs/.vitepress/vitepress/components/navbar/vp-search.vue

```diff
@@ -52,6 +54,12 @@ function update(options: any) {
   }
 }
 
+function convertUrlToCurrentLang(url: string) {
+  return langs.value.some((el) => url.startsWith(`/${el}/`))
+    ? url.replace(/^\/[^/]+/, `/${lang.value}`)
+    : url
```

Comment:
```suggestion
  const regExp = new RegExp(`^\/(${langs.value.join('|')})\\/`)
  return url.replace(regExp, (_, item) => item ? `/${lang.value}/` : undefined)
```

It can also optimize regular matching

Reply:
每个人都有自己的编码风格吧，在没有任何性能影响的前提下，我倒是觉得代码简单易懂才是最重要的

Reply:
时间复杂度 O(logn) 与 O(1) 的区别

Reply:
- 是否包含，是 替换，不是 不处理
- 数组转字符串，字符串转正则，再正则替换

正常情况下正则转换没啥问题，不正常情况下，如果语言包含特殊字符呢？（当然目前来看确实不存在这种情况，以后可能也不会）

我也不是要争论什么，每个人都有自己的实现方式🙂